packages/x-components/src/x-modules/scroll/components/main-scroll.vue

```diff
@@ -0,0 +1,400 @@
+     */
+    @Prop({ default: false, type: Boolean })
+    public useWindow!: boolean;
+
+    /**
+     * Timeout in milliseconds to abort trying to restore the scroll position to the target
+     * element.
+     *
+     * @public
+     */
+    @Prop({ default: 5000 })
+    public restoreScrollTimeoutMs!: number;
+
+    /**
+     * Intersection percentage to consider an element visible.
+     *
+     * @public
+     */
+    @Prop({ default: 0.5 })
+    public threshold!: number;
+
+    /**
+     * Adjusts the size of the scroll container bounds.
+     *
+     * @public
+     */
+    @Prop({ default: '' })
+    public margin!: string;
+
+    /**
+     * If true (default), sets the scroll position to top when an
+     * {@link XEventsTypes.UserAcceptedAQuery} event is emitted.
+     *
+     * @public
+     */
+    @Prop({ type: Boolean, default: true })
+    protected resetOnQueryChange!: boolean;
+
+    /**
+     * The elements that are currently considered visible.
+     *
+     * @internal
+     */
+    protected intersectingElements: HTMLElement[] = [];
+
+    /**
+     * Stores the identifier of the timeout that will consider the scroll failed to restore.
+     *
+     * @internal
+     */
+    protected restoreScrollFailTimeoutId?: number;
+
+    /**
+     * Pending identifier scroll position to restore. If it matches the {@link MainScrollItem.item}
+     * `id` property, this component should be scrolled into view.
+     *
+     * @internal
+     */
+    @State('scroll', 'pendingScrollTo')
+    public pendingScrollTo!: string;
+
+    /**
+     * Intersection observer to determine visibility of the elements.
+     *
+     * @returns An intersection observer to detect elements visibility.
+     * @internal
+     */
+    protected intersectionObserver: IntersectionObserver | null = null;
+
+    /**
+     * Creates an `IntersectionObserver` to detect the first visible elements. Children of this
+     * component should register themselves if they want to be observed.
+     *
+     * @returns The intersection observer.
+     * @public
+     */
+    @XProvide(FirstVisibleItemObserverKey)
+    public get firstVisibleElementObserver(): ScrollVisibilityObserver | null {
+      const observer = this.intersectionObserver;
+      return observer
+        ? {
+            observe: observer.observe.bind(observer),
+            unobserve: element => {
+              this.removeVisibleElement(element);
+              observer.observe(element);
+            }
+          }
+        : null;
+    }
+
+    /**
+     * The first visible element contained in this component.
+     *
+     * @returns The first visible element in this component.
+     * @internal
+     */
+    protected get firstVisibleElement(): HTMLElement | null {
+      return this.intersectingElements.length === 0
+        ? null
+        : this.intersectingElements.reduce((firstVisibleElement, anotherElement) => {
+            // FIXME: This algorithm only takes into account LTR layouts
```

Comment:
As far as I understood, adapting to RTL layouts will only require to change the `left` to `right` in the `anotherElementBounds.left <= firstVisibleElementBounds.left` depending on the value from the snippet.

Reply:
Yep, the only property that you should change is that `left` by a `right`. My problem is how to detect the RTL layout. I don't want to inject the `SnippetConfig` everywhere. Remember that this component might be used without it. And so far the only solution I know is recursively querying the DOM until you find a `dir` attribute.

Reply:
while we don't have elements of 2 rows of width, there is no problem with this, I don't care if I take the first or the last of the row. I only want the vertical position for the scroll.